File: software/examples/ruby/example_callback.rb
#!/usr/bin/env ruby
# -*- ruby encoding: utf-8 -*-

require 'tinkerforge/ip_connection'
require 'tinkerforge/bricklet_multi_touch'

include Tinkerforge

HOST = 'localhost'
PORT = 4223
UID = 'XYZ' # Change to your UID

ipcon = IPConnection.new # Create IP connection
mt = BrickletMultiTouch.new UID, ipcon # Create device object

ipcon.connect HOST, PORT # Connect to brickd
# Don't use device before ipcon is connected

# Register touch_state callback (parameter has unit Lux/10)
mt.register_callback(BrickletMultiTouch::CALLBACK_TOUCH_STATE) do |touch_state|
    s = ''
    if (touch_state & (1 << 12)) == (1 << 12)
        s += 'In proximity, '
    end

    if (touch_state & 0xFFF) == 0
        s += 'No electrodes touched'
    else
        s += 'Electrodes '
        for i in 0..11
            if (touch_state & (1 << i)) == (1 << i)
                s += i.to_s() + ' '
            end
        end
        s += 'touched'
    end

    puts s
    puts
end

puts 'Press key to exit'
$stdin.gets
ipcon.disconnect

File: software/examples/csharp/ExampleCallback.cs
using Tinkerforge;

class Example
{
	private static string HOST = "localhost";
	private static int PORT = 4223;
	private static string UID = "XYZ"; // Change to your UID

	// Callback function for touch state
	static void TouchStateCB(BrickletMultiTouch sender, int touchState)
	{
		string str = "";

		if((touchState & (1 << 12)) == (1 << 12)) {
			str += "In proximity, ";
		}

		if((touchState & 0xfff) == 0) {
			str += "No electrodes touched" + System.Environment.NewLine;
		} else {
			str += "Electrodes ";
			for(int i = 0; i < 12; i++) {
				if((touchState & (1 << i)) == (1 << i)) {
					str += i + " ";
				}
			}
			str += "touched" + System.Environment.NewLine;
		}

		System.Console.WriteLine(str);
	}

	static void Main() 
	{
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickletMultiTouch mt = new BrickletMultiTouch(UID, ipcon); // Create device object

		ipcon.Connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		// Register touchState callback to function TouchStateCB
		mt.TouchState += TouchStateCB;

		System.Console.WriteLine("Press key to exit");
		System.Console.ReadKey();
		ipcon.Disconnect();
	}
}
